server/src/main/java/com/cloud/network/IpAddressManagerImpl.java

```diff
@@ -2238,4 +2245,18 @@ public boolean isIpEqualsGatewayOrNetworkOfferingsEmpty(Network network, String
         }
         return false;
     }
+
+    @Override
+    public boolean isUsageHidden(IPAddressVO ip) {
+        Long networkId = ip.getAssociatedWithNetworkId();
+        if (networkId == null) {
+            networkId = ip.getSourceNetworkId();
+        }
+        if (networkId == null) {
+            // Should not happen
```

Comment:
if it shouldn't happen, isn't it really an exception?

Reply:
@DaanHoogland You're right, I could certainly return a `CloudRuntimeException`.